src/core/patterns/PatternsProvider.tsx

```diff
@@ -25,16 +26,18 @@ const Context = createContext(STATE);
 
 class Provider extends React.Component<PatternsProvider.Props, typeof STATE> {
   componentDidMount() {
-    this.getPatterns();
+    if(this.props.getOnInit) {
+      this.getPatterns();
+    }
   }
 
-  getPatterns = async () => {
+  getPatterns = async (query?: string) => {
```

Comment:
Use type inference here. Instead of nullable query replace this with  query = ' ' without type annotation. Ts will what type query have